hf_olmo/convert_olmo_to_hf.py

```diff
@@ -67,10 +68,16 @@ def write_model(checkpoint_dir: str, ignore_olmo_compatibility: bool = False):
     # For device_map = "auto", etc. the models are loaded in a way that start_prefix is not computed correctly.
     # So, we explicitly store the model with the expected prefix.
 
-    old_model_path = os.path.join(checkpoint_dir, "model.pt")
-    new_model_path = os.path.join(checkpoint_dir, "pytorch_model.bin")
+    if os.path.exists(os.path.join(checkpoint_dir, "model.pt")):
+        old_model_path = os.path.join(checkpoint_dir, "model.pt")
+        state_dict = torch.load(old_model_path, map_location="cpu")
+    elif os.path.exists(os.path.join(checkpoint_dir, "model.safetensors")):
+        old_model_path = os.path.join(checkpoint_dir, "model.safetensors")
+        state_dict = load_file(old_model_path, device="cpu")
```

Comment:
Have you tested this? If you are getting the safetensors file using `unshard.py` then you should probably use `safetensors_file_to_state_dict` instead. See https://github.com/allenai/OLMo/blob/main/docs/Safetensors.md for context.